### ISSUE TITLE
Create MUF "Stylesheet" Library

### ISSUE THREAD

**Issue by tanabi** [theconnstruct/fuzzball-muf#1]
The library will allow us to build a 'color language' and enable people to customize to their own tastes.

At a high level, this library will enable the use of certain "tags" such as 'warning', 'error', 'gui' (for ASCII art elements), and potentially other custom tags because really why not?  These tags will be resolved into ANSI colors that can be overriden in the following fashion:

* Defaults will be in the library
* Settings on #0 override library defaults
* Settings of the MUF can override above defaults
* Settings on the player override all of the above

The proposed structure will be:

```
_styles/tagname:color
_styles/tagname:*bolded color
_styles/tagname:/italic color
_styles/tagname:-overstrike color
_styles/tagname:~blink color
_styles/tagname:*dim color
_styles/tagname:#reverse color
_styles/tagname_underline color
```

These different symbols map to TEXTATTR abilities.  Open question: Do we want to support multiple symbols?  Do we want to have _styles/bg_tagname   to separate backgrounds from foregrounds?

Using these tags should be simple.  We could do some HTML-ish thing like:

```
"This <error>would be an error</error> with a <warning>warning</warning>"
mss_tell  (* mss_tell for MUF/MUCK stylesheet? *)
```
Or maybe to conserve valuable string space, we make it a lil more basic:

```
"This is an <error>error message</>"
````

These would need to be escapable...

````
"This is an \<error>that has been escaped" mss_tell
````

We'd need mss_ versions of all the usual commands, and also just expose the parser function so that people can use it with whatever else they may need to.

**Comment by connstructor** [comment]
> The proposed structure will be:
>
> ```
> _styles/tagname:color
> _styles/tagname:*bolded color
> _styles/tagname:/italic color
> _styles/tagname:-overstrike color
> _styles/tagname:~blink color
> _styles/tagname:*dim color
> _styles/tagname:#reverse color
> _styles/tagname_underline color
> ```

I'm not quite understanding what's being indicated, here.

My initial thought is that a given tag (`error`, `warning`, `info`, `gui`, etc.) would be set to whatever the `textattr` string would be for that element type.

Example:

`_styles/error:bold red`
`_styles/warning:yellow`
`_styles/gui:bold white`

Then using the tag structure within the string to be formatted via `mss_fmt`, `mss_tell`, etc.

**Comment by tanabi** [comment]
> > The proposed structure will be:
> > ```
> > _styles/tagname:color
> > _styles/tagname:*bolded color
> > _styles/tagname:/italic color
> > _styles/tagname:-overstrike color
> > _styles/tagname:~blink color
> > _styles/tagname:*dim color
> > _styles/tagname:#reverse color
> > _styles/tagname_underline color
> > ```
> 
> I'm not quite understanding what's being indicated, here.
> 
> My initial thought is that a given tag (`error`, `warning`, `info`, `gui`, etc.) would be set to whatever the textattr string would be for that element type.
> 
> Example:
> 
> `_styles/error:bold red` `_styles/warning:yellow` `_styles/gui:bold white`
> 
> Then using the tag structure within the string to be formatted via `mss_fmt`, `mss_tell`, etc.

Agreed, when I stepped away for a moment to use the bathroom, I came back and was like, you know the symbol idea is stupid.  Nevermind that, you've got hte right idea.

**Comment by connstructor** [comment]
Awesome, just making sure I'm on the right track. Thanks!
